version: '3.8'

services:
  authelia:
    container_name: authelia
    image: authelia/authelia:latest
    volumes:
      - ./authelia/:/config
    ports:
      - "9091:9091"
    healthcheck:
      test: [ "CMD", "wget", "--spider", "-q", "http://127.0.0.1:9091/" ]
      interval: 30s
      timeout: 10s
      retries: 3
      start_period: 5s
  mailhog:
    container_name: mailhog
    image: mailhog/mailhog:latest
    healthcheck:
      test: [ "CMD", "wget", "--spider", "-q", "http://127.0.0.1:8025/api/v2/messages" ]
      interval: 30s
      timeout: 10s
      retries: 3
      start_period: 5s

  vaultls:
    image: ghcr.io/7ritn/vaultls:dev
    build:
      context: ../
      dockerfile: Containerfile
      args:
        RUN_TESTS: "true"
    depends_on:
      mailhog:
        condition: service_healthy
      authelia:
        condition: service_healthy
    network_mode: "service:authelia"
    volumes:
      - /app/data
    environment:
      - "VAULTLS_LOG_LEVEL=trace"
      - "VAULTLS_API_SECRET=cpCeMZga+H72Ux99y3v+Ht2IDH2NEClh40UTT4Mrxm4="
      - "VAULTLS_URL=http://localhost:5173/"
      - "VAULTLS_MAIL_HOST=mailhog"
      - "VAULTLS_MAIL_PORT=1025"
      - "VAULTLS_MAIL_FROM=test@vaultls.dev"
      - "VAULTLS_CHECK_EXPIRY_INTERVAL=1"
      - "VAULTLS_OIDC_ID=4lMLB1v61sUInLBCjpUvjnmIgXexYKdgfolAahzI0BGYdHxYIsmz4AcdVmsKF4Zk"
      - "VAULTLS_OIDC_SECRET=MSO5oNjfuf0t7Kl4JMoDYNsFaMkw1fiGtpzEO34yqjrJ2usrTyyO8AaK9uqUmpJb"
      - "VAULTLS_OIDC_AUTH_URL=http://127.0.0.1:9091"
      - "VAULTLS_OIDC_CALLBACK_URL=http://127.0.0.1:80/api/auth/oidc/callback"
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
      interval: 30s
      timeout: 10s
      retries: 3
      start_period: 5s

  test:
    build:
      context: ./e2e
      dockerfile: Containerfile
    depends_on:
      vaultls:
        condition: service_healthy
    network_mode: "service:authelia"
    environment:
      - "VAULTLS_URL=http://127.0.0.1/"
    entrypoint: ["pytest", "/app/test_vaultls.py"]